text/0000-global-allocators.md

```diff
@@ -240,9 +252,47 @@ The allocator APIs could be simplified to a more "traditional"
 malloc/calloc/free API at the cost of an efficiency loss when using allocators
 with more powerful APIs.
 
+The global allocator could be an instance of the `Allocator` trait. Since that
+trait's methods take `&mut self`, things are a bit complicated however. The
+allocator would most likely need to be a `const` type implementing `Allocator`
+since it wouldn't be sound to interact with a static. This may cause confusion
```

Comment:
This is not true. Unlike `static mut`s, plain `static`s are perfectly safe to access and can, in fact, maintain state. It's just that all mutation needs to happen via thread safe interior mutability.

In fact, a `static` makes much more sense to me than a `const` &mdash; the latter is a *value* that gets copied everywhere, while the former is a unique object with an identity.